### PR TITLE
increase barrier for available profiles to be ready

### LINKED ISSUE #138999: Powershell opening on Terminal in first instance despite default
- VSCode Version: 1.63.0
- Local OS Version: Windows 10 20H2
- Remote OS Version: N/A
- Remote Extension/Connection Type: SSH/Docker/WSL
- Logs: Unable to find source log that indicates terminal launch

Steps to Reproduce:

1. Open VSCode
2. press CTRL+~ to open terminal

Killing the terminal and executing step 2 will open the default terminal (Git Bash for me)

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

image

### PR DESCRIPTION
**Pull request by meganrogge**
Co-authored-by: Daniel Imms <2193314+Tyriar@users.noreply.github.com>

This PR fixes #138999

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts [hunk 1]
```diff
@@ -109,7 +111,10 @@ export class TerminalProfileService implements ITerminalProfileService {
 		if (profilesChanged || contributedProfilesChanged) {
 			this._availableProfiles = profiles;
 			this._onDidChangeAvailableProfiles.fire(this._availableProfiles);
-			this._profilesReadyBarrier.open();
+			if (!this._profilesReadyBarrier.isOpen()) {
+				this._logService.debug('Took 20 seconds to get available terminal profiles');
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts:115]:
```suggestion
				this._logService.debug('Took over 20 seconds to get available terminal profiles');
```

**Review by Tyriar — DISMISSED** [review]

**Review by Tyriar — DISMISSED** [review]

**Comment by meganrogge** [comment]
tests are failing

File: src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts [hunk 2]
```diff
@@ -67,7 +69,7 @@ export class TerminalProfileService implements ITerminalProfileService {
 		// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
 		// default terminal before launching the first terminal. This isn't expected to ever take
 		// this long.
-		this._profilesReadyBarrier = new AutoOpenBarrier(5000);
+		this._profilesReadyBarrier = new AutoOpenBarrier(20000);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts:72]:
```suggestion
		this._profilesReadyBarrier = new AutoOpenBarrier(15000);
```

File: src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts [hunk 3]
```diff
@@ -109,7 +111,10 @@ export class TerminalProfileService implements ITerminalProfileService {
 		if (profilesChanged || contributedProfilesChanged) {
 			this._availableProfiles = profiles;
 			this._onDidChangeAvailableProfiles.fire(this._availableProfiles);
-			this._profilesReadyBarrier.open();
+			if (!this._profilesReadyBarrier.isOpen()) {
+				this._logService.debug('Took over 20 seconds to get available terminal profiles');
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts:115]:
```suggestion
				this._logService.debug('Took 20 seconds to get available terminal profiles');
```

File: src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts (hunk 1 above)
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProfileService.ts:115]: I think I got confused here, if it's not open it means it's less than 20 seconds. Opps, let's just revert this and do without the warning

**Review by Tyriar — CHANGES_REQUESTED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**